[Ou Jin Bin] iP

kelvinou01 opened the pull request:
### Roger
> "Only failure forget their task" - Confucius

Roger helps you remember your tasks. It's,

- text-based
- polite
- super fast

All you need to do is 

1. [Download it](https://github.com/kelvinou01/ip/releases/tag/A-Jar)
2. Add your tasks
3. Let it manage your tasks for you 😩💦

And its **FREE**!

Features:

- [x] Managing tasks
- [ ] Reminders you of your failures (coming soon)

------
For Java programmers, here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

kelvinou01 committed:
Add .gitignore

kelvinou01 committed:
Rename Duke to Roger, and add core functionality

kelvinou01 committed:
Add  and

kelvinou01 committed:
Add the Task class

kelvinou01 committed:
Add mark/unmark functionality, and refactor to use the Task class

kelvinou01 committed:
Add handling for invalid input for `mark` or `unmark`

kelvinou01 committed:
Add Deadline class

kelvinou01 committed:
Add Event class

kelvinou01 committed:
Add ToDo class

kelvinou01 committed:
Modify `list()` behaviour when the task list is empty

kelvinou01 committed:
Integrate ToDo, Deadline, and Event into Roger

kelvinou01 committed:
Fixed error message

kelvinou01 committed:
Change logo to only use ASCII characters, to fix issue with runtest.sh

kelvinou01 committed:
Add automated text UI testing

kelvinou01 committed:
Add RogerInvalidInputException class

kelvinou01 committed:
Integrate RogerInvalidInputException and add exception handling

kelvinou01 committed:
Update automated text UI testing

kelvinou01 committed:
Fix formatting

kelvinou01 committed:
Add delete task functionality

kelvinou01 committed:
Update automated text UI testing

kelvinou01 committed:
Add toStorageFormat() for Deadline, Event, ToDo, and Task

kelvinou01 committed:
Add task saving and loading to hard disk

kelvinou01 committed:
Change `Deadline::date` and `Event::date` to use java.time.LocalDate instead of String

kelvinou01 committed:
Refactor command handler methods to take in arguments instead of whole line, and integrate LocalDate functionality.

kelvinou01 committed:
Merge branch 'branch-Level-8'

kelvinou01 committed:
Task class: change to abstract

kelvinou01 committed:
Add Storage and StorageParser classes.

File storage logic is spread out between the individual task classes and Roger itself.

Encapsulate local task storage into the Storage class, and encapsulate the storage format into the StorageParser class.

To obey SRP.

kelvinou01 committed:
StorageParser class: fix format string

kelvinou01 committed:
StorageParser class: add exception handling

kelvinou01 committed:
Add TaskList class

We are using List<Task> as in-program task storage.

Now, we replace List<Task> with TaskList class.

It is cleaner to encapsulate actions that we can perform with the list of tasks into a single class.

kelvinou01 committed:
Add Ui class

kelvinou01 committed:
TaskList class: Add filtered tasks display.

There is no good way to display filtered tasks.

Add methods to filter the tasklist, get display strings of the tasks, and get task number for displaying.

This way, users of the TaskList class can
* display individual tasks with task number, and
* diplay ffiltered lists with task numbers,
without having access to the internal List<Task>

kelvinou01 committed:
Add Parser and Command classes.

Parsing is being handled by the main method and handle<Command> methods, while command execution is handled by <command> methods.

Encapsulate parsing user input within a single Parser class, and the various commands in various Command subclasses.

This ensures that the logic is grouped into single responsibility units.

kelvinou01 committed:
Refactor everything into packages

kelvinou01 committed:
Add Gradle support

kelvinou01 committed:
Fix storage parser args parsing

kelvinou01 committed:
Add unit tests

kelvinou01 committed:
Add JavaDocs

kelvinou01 committed:
Refactored code to conform to style guide.

kelvinou01 committed:
TaskList class: Add search method

kelvinou01 committed:
Add `find` command

Users do not have a way to find tasks quickly.

Add a find command that filters tasks based on a query string.

Makes it easy to find tasks that users are looking for.

kelvinou01 committed:
Merge branch-A-JavaDoc

kelvinou01 committed:
Merge branch-A-CodingStandard

kelvinou01 committed:
Refactor StorageParser

kelvinou01 committed:
Fix wrongly hard coded file path

kelvinou01 committed:
Add CheckStyle plugin

kelvinou01 committed:
Add JavaFX UI

Roger was a CLI program

Major changes include:
- Add JavaFX FXML files and controllers
- Change Roger from a CLI program into a backend, which provides the `getResponse(input)` method.
- Encapsulate command results into `CommandResult`
However, the core functionality of Roger remains unchanged.

A UI is much nicer to use than a CLI program.

kelvinou01 committed:
Merge branch 'branch-A-CheckStyle'

# Conflicts:
#	src/main/java/roger/commands/ExitCommand.java

kelvinou01 committed:
Refactor code to conform to style guide

craeyeons commented on src/main/java/roger/ui/Parser.java:
```diff
@@ -0,0 +1,182 @@
+ * Encapsulates logic for parsing user input.
+ */
+public class Parser {
+    private final String COMMAND_LIST = "list";
+    private final String COMMAND_MARK = "mark";
+    private final String COMMAND_UNMARK = "unmark";
+    private final String COMMAND_FIND = "find";
+    private final String COMMAND_ADD_TODO = "todo";
+    private final String COMMAND_ADD_DEADLINE = "deadline";
+    private final String COMMAND_ADD_EVENT = "event";
+    private final String COMMAND_DELETE = "delete";
+    private final String COMMAND_EXIT = "bye";
```
I like how you use constants to define commands to avoid error.

craeyeons commented on src/main/java/roger/ui/controllers/MainWindow.java:
```diff
@@ -0,0 +1,60 @@
+import javafx.scene.image.Image;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.VBox;
+import roger.ui.Response;
```
I think i'd like to see a line between the two import statements, so I can clearly see a line between javafx imports and roger inputs.

craeyeons commented on src/test/java/roger/ParserTest.java:
```diff
@@ -0,0 +1,29 @@
+import roger.exceptions.RogerInvalidInputException;
+import roger.ui.Parser;
+
+import static org.junit.jupiter.api.Assertions.*;
```
I think I'd prefer imported classes to be listed explicitly.

craeyeons commented on src/test/java/roger/StorageParserTest.java:
```diff
@@ -0,0 +1,25 @@
+import roger.tasks.Event;
+import roger.tasks.ToDo;
+
+import static org.junit.jupiter.api.Assertions.*;
```
I think I'd prefer imported classes to be listed explicit

craeyeons commented on src/main/java/roger/tasks/TaskList.java:
```diff
@@ -0,0 +1,169 @@
+    }
+
+    /**
+     * Create a TaskList, and initialise it with a list of tasks.
```
I think US spelling will be more appropriate, as it caters to a wider audience. Hence I think it should be initialize instead of initialise.

craeyeons reviewed (commented):
Overall good job! Just some nits to fix.

kelvinou01 committed:
Add greet functionality

kelvinou01 committed:
Refactor to fix style violations

kelvinou01 committed:
Roger class: Add assertions

No assertions are used to document assumptions made.

We need to make the assumptions explicit via assertions.

Now, we add assertions to methods that use various fields.

This way, the assertion will catch any wrong modifications to the code which may cause NullPointerException

kelvinou01 committed:
Storage class: Add assertions

No assertions are used to document assumptions made.

We need to make the assumptions explicit via assertions.

Now, we add assertions to methods that use various fields.

This way, the assertion will catch any wrong modifications to the code which may cause NullPointerException

kelvinou01 committed:
Refactor code to conform to code quality guide

Code conforms largely to code quality guide, but there is still room for improvement.

Add improvements like extracting magic strings into constants, and extracting methods to SLAP hard.

This improves code readability and maintainability.

kelvinou01 committed:
Merge pull request #1 from kelvinou01/branch-A-Assertions

branch-A-assertions

kelvinou01 committed:
Merge pull request #2 from kelvinou01/branch-A-CodeQuality

Refactor code to conform to code quality guide

kelvinou01 committed:
Add Github Actions CI workflow

kelvinou01 committed:
Add Github Actions CI workflow

kelvinou01 committed:
Move `date` from `Event` and `Deadline` into `Task`.

Utilise polymorphism better.

kelvinou01 committed:
Add Schedule command

kelvinou01 committed:
Refactor code to conform to style guide

kelvinou01 committed:
Make exception message more descriptive

kelvinou01 committed:
Fix program crash upon receiving doubly-spaced input

kelvinou01 committed:
Add tests for Response class

kelvinou01 committed:
Add test for TaskStub class

kelvinou01 committed:
Add testing stubs

kelvinou01 committed:
Move ParserTest into ui

kelvinou01 committed:
Move StorageParserTest into storage

kelvinou01 committed:
Add tests for Roger class

kelvinou01 committed:
Add tests for TaskList

kelvinou01 committed:
Refactor tests to conform to style guide

kelvinou01 committed:
Merge branch 'branch-A-MoreTesting' of https://github.com/kelvinou01/ip

kelvinou01 committed:
Add product screenshot

kelvinou01 committed:
Add user guide

kelvinou01 committed:
Change build.gradle to conform exactly to SE-EDU tutorial

kelvinou01 committed:
Add potential fix for jar file not running

See: https://github.com/nus-cs2103-AY2223S1/forum/issues/169

kelvinou01 committed:
Remove unused dependencies

kelvinou01 committed:
Fix build not working on windows

kelvinou01 committed:
Fix Main not referencin the correct `Application.class`

kelvinou01 committed:
Refactor code

kelvinou01 committed:
Handle deleting non-existent tasks

kelvinou01 committed:
Update Javadocs

kelvinou01 committed:
Change location to save and load from
